lib/dhcp/client/client.go

```diff
@@ -85,17 +85,21 @@ func (c *client) SetTimeout(t time.Duration) {
 	c.timeout = t
 }
 
-func withRetry(op func() error) error {
+func withRetry(name string, op func() error) error {
 	defer trace.End(trace.Begin(""))
 
 	for {
 		if err := op(); err != nil {
-			if errno, ok := err.(syscall.Errno); !ok || errno != syscall.EAGAIN {
-				return err
+			if errno, ok := err.(syscall.Errno); ok {
```

Comment:
not that important but a note would be nice here saying that go runtime sets SA_RESTART for syscalls and retires them automatically if they interrupted.

I believe the reason we need to do this here because we set a timeout on the socket and signal man page (http://man7.org/linux/man-pages/man7/signal.7.html) says following;

```
 The following interfaces are never restarted after being interrupted
       by a signal handler, regardless of the use of SA_RESTART; they always
       fail with the error EINTR when interrupted by a signal handler:

       * "Input" socket interfaces, when a timeout (SO_RCVTIMEO) has been
         set on the socket using setsockopt(2): accept(2), recv(2),
         recvfrom(2), recvmmsg(2) (also with a non-NULL timeout argument),
         and recvmsg(2).

       * "Output" socket interfaces, when a timeout (SO_RCVTIMEO) has been
         set on the socket using setsockopt(2): connect(2), send(2),
         sendto(2), and sendmsg(2).
...
```